libafl/src/executors/timeout.rs

```diff
@@ -151,22 +168,38 @@ where
         unsafe {
             let data = &mut GLOBAL_STATE;
             write_volatile(
-                &mut data.timer_queue,
-                &mut self.timer_queue as *mut _ as *mut c_void,
+                &mut data.tp_timer,
+                &mut self.tp_timer as *mut _ as *mut c_void,
             );
-            let code = CreateTimerQueueTimer(
-                &mut self.ph_new_timer,
-                &self.timer_queue,
-                Some(self.executor.timeout_handler()),
-                &mut GLOBAL_STATE as *mut _ as *mut c_void,
-                self.milli_sec,
-                0,
-                WORKER_THREAD_FLAGS::default(),
+            write_volatile(
+                &mut data.critical,
+                &mut self.critical as *mut _ as *mut c_void,
             );
-            if !code.as_bool() {
-                return Err(Error::Unknown("CreateTimerQueue failed.".to_string()));
-            }
+            write_volatile(
+                &mut data.timeout_input_ptr,
+                &mut data.current_input_ptr as *mut _ as *mut c_void,
+            );
+            let tm: i64 = -1 * self.milli_sec * 10 * 1000;
+            let mut ft = FILETIME::default();
+            ft.dwLowDateTime = (tm & 0xffffffff) as u32;
+            ft.dwHighDateTime = (tm >> 32) as u32;
+
+            EnterCriticalSection(&mut self.critical);
+            write_volatile(&mut data.in_target, 1);
```

Comment:
There's no need for a volatile write if you are inside of the critical section, I think? It's only important to have a compiler barrier after

Reply:
yes, true.